### PR TITLE
software: store relative paths for masks in MVS scene and fix global space IDs

### PR DESCRIPTION
**Pull request by 4CJ7T**
image

### CODE REVIEW

**Comment by csparker247** [comment]
It's been awhile, so I'm just bumping this. @pmoulon (or some other maintainer) can we get this reviewed and merged? This fixes some issues I'm having with masks and OpenMVS.

**Comment by 4CJ7T** [comment]
@Sanheiii Is it alright if I remove global IDs?

**Comment by Sanheiii** [comment]
It's been a long time since my last commit, I don't remember the details of it, you can check the code to decide if you want to remove it or not.

**Comment by 4CJ7T** [comment]
```json
{
  "name": "images\\DSC09972.JPG",
  "mask_name": "images\\DSC09972.mask.png",
  "platform_id": 0,
  "camera_id": 0,
  "pose_id": 27,
  "id": 27
},
{
  "name": "images\\DSC09973.JPG",
  "mask_name": "images\\DSC09973.mask.png",
  "platform_id": 1,
  "camera_id": 0,
  "pose_id": 0,
  "id": 28
}
```
@pmoulon good to go chef.

File: src/software/SfM/export/InterfaceMVS.h [hunk 1]
```diff
@@ -514,14 +521,40 @@ struct Interface
...
 		uint32_t poseID; // ID of the pose of the associated platform
-		uint32_t ID; // ID of this image in the global space (optional)
+		uint32_t ID; // image global-ID, ex. the ID given outside the current scene, like the index in the full list of image files (optional)
+		float minDepth; // minimum depth of the points seen by this image (optional)
+		float avgDepth; // average depth of the points seen by this image (optional)
+		float maxDepth; // maximum depth of the points seen by this image (optional)
+		std::vector<ViewScore> viewScores; // list of view scores for this image (optional)
 
-		Image() : platformID(NO_ID), cameraID(NO_ID), poseID(NO_ID), ID(NO_ID) {}
+		Image() : platformID(NO_ID), cameraID(NO_ID), poseID(NO_ID), ID(NO_ID), minDepth(0), avgDepth(0), maxDepth(0) {}
```
Comment (pmoulon) [inline src/software/SfM/export/InterfaceMVS.h:557]: @4CJ7T  I see more information here than what exists in https://github.com/cdcseacave/openMVS/blob/master/libs/MVS/Interface.h#L524 is it normal

**Comment by pmoulon** [comment]
@4CJ7T Please let me know why this https://github.com/openMVG/openMVG/pull/2193#discussion_r1374048384 does not match OpenMVS repo

File: src/software/SfM/export/InterfaceMVS.h (hunk 1 above)
Comment (4CJ7T) [reply src/software/SfM/export/InterfaceMVS.h:557]: It is the latest interface.h from the develop branch.

Comment (4CJ7T) [reply src/software/SfM/export/InterfaceMVS.h:557]: Compatible with the main branch.

Comment (4CJ7T) [reply src/software/SfM/export/InterfaceMVS.h:557]: @pmoulon

**Comment by pmoulon** [comment]
Thank you

**Pull request merged**